### PR TITLE
RDM-7635: Migrate Stub Service to OIDC

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-7635

### Change description ###

Implement OpenID changes in Stub service

**Does this PR introduce a breaking change?** (check one with "x")

```
[ X] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/ccd/test/stubs/service/token/JWTokenGenerator.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                                       Map<String, Object> claims) {
+        final SignatureAlgorithm signatureAlgorithm = SignatureAlgorithm.RS256;
+
+        final long nowMillis = System.currentTimeMillis();
+        final Date now = new Date(nowMillis);
+
+        final Key signingKey = KeyUtil.getPrivateKey(privateKeyFileName);
+
+        final JwtBuilder builder = Jwts.builder()
+            .setId(UUID.randomUUID().toString())
```
Comment (smathangi) [inline src/main/java/uk/gov/hmcts/reform/ccd/test/stubs/service/token/JWTokenGenerator.java]: i guess it should map with wiremock jwks response. So, better use some fixed value.

File: src/main/java/uk/gov/hmcts/reform/ccd/test/stubs/service/token/JWTokenParser.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import io.jsonwebtoken.Jwts;
+import java.security.Key;
+
+public final class JWTokenParser {
+
+    private JWTokenParser() {
+    }
+
+    public static Claims parseToken(String publicKeyFileName, String jwt) {
+        final Key signingKey = KeyUtil.getPublicKey(publicKeyFileName);
```
Comment (smathangi) [inline src/main/java/uk/gov/hmcts/reform/ccd/test/stubs/service/token/JWTokenParser.java]: where do we use this ?

File: src/main/java/uk/gov/hmcts/reform/ccd/test/stubs/service/token/KeyUtil.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        } catch (final InvalidKeySpecException ex) {
+            throw new RuntimeException(
+                "Error reading public key '" + filename + "'.", ex
+            );
+        }
+    }
+
+    public static PrivateKey getPrivateKey(String filename) {
+        try {
+            final byte[] keyBytes = Files.readAllBytes(Paths.get(ClassLoader.getSystemResource(filename).toURI()));
```
Comment (smathangi) [inline src/main/java/uk/gov/hmcts/reform/ccd/test/stubs/service/token/KeyUtil.java]: Docker has some issue reading files using  `getSystemResource`. Please have a look and test using ccd-docker management-web UI login

File: src/main/java/uk/gov/hmcts/reform/ccd/test/stubs/service/token/JWTokenGenerator.java (hunk 1 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/reform/ccd/test/stubs/service/token/JWTokenGenerator.java]: private key should be kept secret right.. we don't need to expose that outside.. if we provide the json file then we can access that from outside...
wiremock response should include only public keys... our existing /o/jwks endpoint exposing public key only....

File: src/main/java/uk/gov/hmcts/reform/ccd/test/stubs/service/token/JWTokenParser.java (hunk 2 above)
Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/reform/ccd/test/stubs/service/token/JWTokenParser.java]: we are not using this... but will be useful for testing...

File: Generate Keys.txt [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Generate Keys
```
Comment (mario-paniccia) [inline Generate Keys.txt]: if this is a manual step can we update the Readme?

Comment (kiran-yenigala-hmcts) [reply Generate Keys.txt]: This is not needed for the changed implementation

Comment (smathangi) [reply Generate Keys.txt]: removed

**Pull request merged**